GPyOpt/optimization/acquisition_optimizer.py

```diff
@@ -62,9 +65,9 @@ def optimize(self, f=None, df=None, f_df=None, duplicate_manager=None):
 
         ## --- Selecting the anchor points and removing duplicates
         if self.type_anchor_points_logic == max_objective_anchor_points_logic:
-            anchor_points_generator = ObjectiveAnchorPointsGenerator(self.space, random_design_type, f)
+            anchor_points_generator = ObjectiveAnchorPointsGenerator(self.space, random_design_type, f, user_def_dist=self.user_def_dist)
```

Comment:
I think there may be some changes that can be made, but options are ultimately limited by a few things:
1. `user_def_dist` needs to be passed to `GPyOpt.util.general.initial_design()` whenever it is called.
2.  This can be achieved by passing `user_def_dist` as an argument to `GPyOpt.optimization.anchor_points_generator.AnchorPointsGenerator.get()` (instead of setting it as a class variable in `AnchorPointsGenerator.__init__()`) or making `user_def_dist` a class variable of `GPyOpt.optimization.acquisition_optimizer.ContextManager` (though it may not be very relevant to place here).
3. Whether or not `user_def_dist` is a class variable of `AnchorPointsGenerator`, the anchor points generator is re-instantiated  every time `GPyOpt.optimizer.acquisition_optimizer.AcquisitionOptimizer.optimize()` is called. In order for `user_def_dist` to persist over multiple calls of optimize, `user_def_dist`  should either be passed in as an argument of `AcquisitionOptimizer.optimize()`, included as a class variable of `AcquisitionOptimizer`, or as a class variable of `ContextManager`. Including `user_def_dist` as a class variable of `AcquisitionOptimizer` seems like the best solution without making bigger changes.

How about this?
&nbsp;&nbsp;&nbsp;&nbsp;A. In `AcquisitionOptimizer.__init__()`, introduce `self.user_def_dist = None` or `self.user_def_dist = kwargs.get('user_def_dist', None)`.
&nbsp;&nbsp;&nbsp;&nbsp;B. `user_def_dist` may be passed as a kwarg to `GPyOpt.methods.bayesian_optimization.BayesianOptimization` which is then passed in to the instantiation of `GPyOpt.util.arguments_manager.ArgumentsManager` with the rest of the kwargs. This way `BayesianOptimization.__init__()` does not need to explicitly pass `user_def_dist` to `AcquisitionOptimizer.__init__()`. (However, `BayesianOptimization` will still need to get `user_def_dist` from kwargs in `BayesianOptimization._init_design_chooser()` since this function calls `initial_design()`.)
&nbsp;&nbsp;&nbsp;&nbsp;C. In `ArgumentsManager.acquisition_creator()`, set `self.acquisition_optimizer.user_def_dist = self.kwargs.get('user_def_dist', None)` . This would satisfy 3. from above.
&nbsp;&nbsp;&nbsp;&nbsp;D. When `AcquisitionOptimizer.optimize()` is called, pass `user_def_dist` as an argument to `anchor_points_generator.get()`. If modifying the arguments to `AnchorPointsGenerator.get()` is undesirable, I can alternatively include `user_def_dist` as a class variable in `ContextManager` (This would change the procedure above) as discussed in 3.
&nbsp;&nbsp;&nbsp;&nbsp;E. From `AnchorPointsGenerator.get()`, `user_def_dist` is passed into `initial_design()`.

Reply:
@apaleyes What do you think? shall we merge?